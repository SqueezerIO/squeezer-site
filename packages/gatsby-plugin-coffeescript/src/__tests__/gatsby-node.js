const {
  resolvableExtensions,
  modifyWebpackConfig,
  preprocessSource,
} = require(`../gatsby-node`)

describe(`gatsby-plugin-coffeescript`, () => {
  it(`contains coffee script extensions`, () => {
    expect(resolvableExtensions()).toMatchSnapshot()
  })

  it(`modifies webpack config with cofeescript extensions`, () => {
    const spy = jest.fn()
    const config = {
      loader(...args) {
        spy(...args)
      },
    }

    modifyWebpackConfig({ config })

    expect(spy).toHaveBeenCalledTimes(resolvableExtensions().length)
  })

  describe(`pre processing`, () => {
    it(`returns null if non-coffeescript file`, () => {
      expect(
        preprocessSource({
          filename: `test.js`,
          contents: `alert('hello');`,
        })
      ).toBe(null)
    })

    it(`transforms .coffee files`, () => {
      expect(
        preprocessSource(
          {
            filename: `test.coffee`,
            contents: `alert "I knew it!" if elvis?`,
          },
          {}
        )
      ).toMatchSnapshot()
    })

    it(`transforms .cjsx files`, () => {
      expect(
        preprocessSource(
          {
            filename: `test.cjsx`,
            contents: `
          React = require('react');

          module.exports = class extends React.Component {
            render: ->
              <div>
                <h1>Hello World</h1>
              </div>
          }
        `,
          },
          {}
        )
      ).toMatchSnapshot()
    })
  })
})
