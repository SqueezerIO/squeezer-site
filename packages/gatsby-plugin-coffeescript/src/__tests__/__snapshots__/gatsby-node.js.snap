// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-coffeescript contains coffee script extensions 1`] = `
Array [
  ".coffee",
  ".cjsx",
]
`;

exports[`gatsby-plugin-coffeescript pre processing transforms .cjsx files 1`] = `
"(function() {
  var React;

  React = require('react');

  module.exports = class extends React.Component({
      render: function() {
        return React.createElement(\\"div\\", null, React.createElement(\\"h1\\", null, \\"Hello World\\"));
      }
    }) {};

}).call(this);
"
`;

exports[`gatsby-plugin-coffeescript pre processing transforms .coffee files 1`] = `
"(function() {
  if (typeof elvis !== \\"undefined\\" && elvis !== null) {
    alert(\\"I knew it!\\");
  }

}).call(this);
"
`;
